File: cpp/solutions/contains_duplicate.cpp
#include "contains_duplicate.h"
#include <unordered_set>

bool ContainsDuplicate::hasDuplicate(std::vector<int>& nums) {
    std::unordered_set<int> seen;
    for (int n : nums) {
        if (seen.count(n)) {
            return true;
        }
        seen.insert(n);
    }
    return false;
};

File: javaneetcode/ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int n : nums) {
            if (seen.contains(n)) {
                return true;
            }
            seen.add(n);
        }
        return false;
    }
}